module_analysis/post_init_hook.py

```diff
@@ -2,13 +2,23 @@
 # @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
+import os
 from odoo import SUPERUSER_ID, api
 
 
 def analyse_installed_modules(cr, registry):
     with api.Environment.manage():
         env = api.Environment(cr, SUPERUSER_ID, {})
-        installed_modules = env["ir.module.module"].search(
-            ["|", ("state", "=", "installed"), ("name", "=", "module_analysis")]
-        )
+        if os.environ.get("CI"):
+            installed_modules = env["ir.module.module"].search(
+                [
+                    "|",
+                    ("state", "=", "installed"),
+                    ("name", "in", ("base", "base_automation", "module_analysis")),
+                ]
+            )
+        else:
+            installed_modules = env["ir.module.module"].search(
+                ["|", ("state", "=", "installed"), ("name", "=", "module_analysis")]
+            )
```

Comment:
I don't get that change. 
AFAIU, if it we are in a CI context, the analysis is done on more modules. Is it desired ?

Reply:
@legalsylvain not my change!  (and yes I don't get it either)
https://github.com/OCA/server-tools/pull/2085/commits/58c2304a349c3899a41d351b5cea8b5440db5054#diff-91be04b85dd85fbd3a212c85ecb10524d6239d5a204189493aa46df065b0562fL12

![Spider_Man_meme](https://user-images.githubusercontent.com/16926/215213373-11df8335-469a-44e1-a46f-c5704524f678.jpg)

Reply:
Ahaha ! 

No. 
- I'm talking about that commit https://github.com/OCA/server-tools/pull/2085/commits/204afc6a87a67cf66f3b015749f192fd91a3b7a5 and it's your.
- regarding the "["|", ("state", "=", "installed"), ("name", "=", "module_analysis")]" I don't remember precisely, but I guess that when installing module_analysis module, in the post_init_hook step, the module is not in the installed state. (I think the module is set as installed after the post_init_hook).

Reply:
Instead of adding test, I think we could just remove the call of analyse_installed_modules in the test.
I mean, it's useless to add a test, if it's to do a if==CI inside. Don't you think ?

Reply:
@legalsylvain in fact the analysis was triggered both by the test and by the post_init hook (and that is why my previous change failed to skip the analysis in the CI context). So I kept an if os.environ.get("CI") test in the _analyse_code method in my last commit. As you can see it succeed in keeping the analysis only for 3 modules max on the CI (and it's probably better to still limit to these 3 modules in the test if your run the test without setting your local CI environment variable.